ui/src/app/layout/createtopic/datasettings/data-settings.component.ts

```diff
@@ -0,0 +1,177 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { TopicType } from '../../../shared/services/topics.service';
+import { KafkaEnvironment } from '../../../shared/services/environments.service';
+import * as moment from 'moment';
+import { ChangeContext, Options } from '@angular-slider/ngx-slider';
+
+type CleanUpStrategy = 'compact' | 'delete';
+
+type Criticality = 'NORMAL' | 'CRITICAL';
+
+type MessagesPerDay = 'LOW' | 'NORMAL' | 'LARGE' | 'VERY_LARGE';
+
+type MessagesSize = 'VERY_SMALL' | 'SMALL' | 'NORMAL' | 'LARGE' | 'VERY_LARGE';
+
+export interface TopicSettingsData {
+
+    subscriptionApprovalRequired: boolean;
+
+    cleanUpStrategy: CleanUpStrategy[];
+
+    compactionTimeMillis: number;
+
+    retentionTimeMillis: number;
+
+    criticality: Criticality;
+
+    messagesPerDay: MessagesPerDay;
+
+    messagesSize: MessagesSize;
+
+}
+
+@Component({
+    selector: 'app-data-settings',
+    templateUrl: './data-settings.component.html',
+    styleUrls: ['./data-settings.component.scss']
+})
+export class DataSettingsComponent {
+
+    @Output() setTopicSettings = new EventEmitter<TopicSettingsData>();
+
+    @Input() topicType: TopicType;
+
+    @Input() selectedEnvironment: KafkaEnvironment;
+
+    @Input() showSubscriptionApprovalRequired: boolean;
+
+    subscriptionApprovalRequired = false;
+
+    activeButtons = { logCompaction: false, deletion: false };
+
+    selectedCleanUpStrategy: CleanUpStrategy[] = ['delete'];
+
+    compactionTime = 24;
+
+    compactionTimeUnit = 'h';
+
+    retentionTime = 7;
+
+    retentionTimeUnit = 'd';
+
+    criticalityType: Criticality = 'NORMAL';
+
+    dataIntervals: string[] = ['< 1.000', '< 100.000', '< 1.000.000', '> 1.000.000'];
```

Comment:
This is german number notation. Better use <1K, <100K, <1M, >1M, or use translation. (English would e.g. be 1,000,000)